raiden/tests/scenarios/long-running.yaml

```diff
@@ -0,0 +1,319 @@
+            # Make a deposit bigger then the deposited amount
+            - deposit: {from: 2, to: 4, total_deposit: 150}
+            - wait: 10
+      - parallel:
+          name: "Verify the channels"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 101, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+            # New channel
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 150, state: "opened"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 0, state: "opened"}
+      - serial:
+          name: "Continue building the network"
+          tasks:
+            # A opens a channel with E
+            - open_channel: {from: 0, to: 4, total_deposit: 10, expected_http_status: 201}
+            - assert: {from: 0, to: 4, total_deposit: 10, balance: 10, state: "opened"}
+            # E deposits 25 tokens onto the A - E channel
+            - deposit: {from: 4, to: 0, total_deposit: 25}
+            - wait: 20
+            # A performs a payment to E of 50 tokens (path A<->B<->C<->E)
+            - transfer: {from: 0, to: 4, amount: 50, expected_http_status: 200}
+            - wait: 5
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 50, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 150, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 50, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 150, state: "opened"}
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 100, state: "opened"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 50, state: "opened"}
+            # C sends all of its tokens to B (one transfer)
+            - transfer: {from: 2, to: 1, amount: 150, expected_http_status: 200}
+            - wait: 5
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 0, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 200, state: "opened"}
+            # C tries to send another > 25 tokens payment to B (fail no route with enough capacity)
+            - transfer: {from: 2, to: 1, amount: 150, expected_http_status: 409}
+            - wait: 5
+      - serial:
+          name: "Multiple multi hop payments"
+          repeat: 10
+          tasks:
+            # C sends 10 payments of 1 token to B by using the C <-> E <-> A <-> B route
+            - transfer: {from: 2, to: 1, amount: 1}
```

Comment:
Also as explained:
1. node (2) has a direct channel with node(1) so the comment above does not apply unless you close the channel between (2) and (1)
2. node (2) has zero balance which means that it can't do any transfers from (2) to (1). 

If you close the channel from (2) to (1) then (2) will send the transfer through (4) where (2) has a balance of 100.

Reply:
Thanks for taking the time to respond review this Rakan, I have write above my arguments of why I think that transfer was supposed to work.

Reply:
Also one more thing that I have to mention is that it used to pass that transfer, and not get stuck there.